File: 프로그래머스/0/120907. OX퀴즈/OX퀴즈.cpp
#include <string>
#include <vector>
#include <sstream>
using namespace std;

vector<string> solution(vector<string> quiz) {
    vector<string> answer;
    vector<string> arr;
    string temp;
    
    for(int i = 0; i < quiz.size(); i++)
    {
        stringstream ss(quiz[i]);
        while(getline(ss,temp,' '))
            arr.push_back(temp);
        
        int num1 = stoi(arr[0]);
        int num2 = stoi(arr[2]);
        int result = stoi(arr[4]);
        
        if(arr[1] == "-")
            answer.push_back(num1 - num2 == result ? "O" : "X");
        else
            answer.push_back(num1 + num2 == result ? "O" : "X");
        
        arr.clear();
    }
    return answer;
}

File: 프로그래머스/0/120907. OX퀴즈/OX퀴즈.cs
using System;

public class Solution {
    public string[] solution(string[] quiz) {
        string[] answer = new string[quiz.Length];
        
        for(int i = 0; i < quiz.Length; i++)
        {
            string[] arr = quiz[i].Split(' ');
            int num1 = Int32.Parse(arr[0]);
            int num2 = Int32.Parse(arr[2]);
            int result = Int32.Parse(arr[4]);
            
            if(arr[1] == "-")
                answer[i] = num1 - num2 == result ? "O" : "X";
            else
                answer[i] = num1 + num2 == result ? "O" : "X";
        }
        
        return answer;
    }
}